### PR TITLE
chore: rename use_gpu to disable_gpu in TorchAlgo

### PR DESCRIPTION
**Pull request by ThibaultFy**
closes FL-1677

## Companion

- https://github.com/Substra/substra-documentation/pull/436
-  https://github.com/Substra/substrafl/pull/241

### CODE REVIEW

File: tests/algorithms/pytorch/test_base_algo.py [hunk 1]
```diff
@@ -198,18 +198,18 @@ def __init__(self):
...
                     index_generator=nig,
-                    use_gpu=use_gpu,
+                    disable_gpu=disable_gpu,
                 )
-            if use_gpu:
-                assert self._device == torch.device("cuda")
-            else:
+            if disable_gpu:
                 assert self._device == torch.device("cpu")
+            else:
```
Comment (SdgJlbl) [inline tests/algorithms/pytorch/test_base_algo.py:205]:
```suggestion
            elif torch.cuda.is_available():
```

File: tests/algorithms/pytorch/test_fed_avg.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def __init__(
                 model=perceptron,
                 index_generator=nig,
                 dataset=numpy_torch_dataset,
-                use_gpu=False,
+                disable_gpu=True,
```
Comment (SdgJlbl) [inline tests/algorithms/pytorch/test_fed_avg.py:49]: I don't think this is required 🤔 
and same for the tests of other implementations

**Review by SdgJlbl — COMMENTED** [review]
Thanks for the PR, I have a few comments related to the fact that this change (which I think is a good one) is a bit more than just a renaming, but is offering the option to not use the GPU even if it is available, else, `substrafl` will default to the available device in a smart way.

File: tests/algorithms/pytorch/test_base_algo.py [hunk 3]
```diff
@@ -499,7 +499,7 @@ def test_gpu(
         dependencies=algo_deps,
         experiment_folder=session_dir / "experiment_folder",
         clean_models=False,
-        name=f'Testing the GPU - strategy {strategy_class.__name__}, running  on {"cuda" if use_gpu else "cpu"}',
+        name=f'Testing the GPU - strategy {strategy_class.__name__}, running  on {"cpu" if disable_gpu else "cuda"}',
```
[the remark this answers is not in the document]
Comment (ThibaultFy) [reply tests/algorithms/pytorch/test_base_algo.py:502]: I agree but this test is suppose to work even if cuda is not available ^^" I'll dig a bit more to see if there is a better way to do this

[the remark this answers is not in the document]
Comment (ThibaultFy) [reply tests/algorithms/pytorch/test_base_algo.py]: Ok I was wrong, I did the changes thanks :)

**Review by SdgJlbl — APPROVED** [review]
Thanks for the PR 🙏

**Pull request merged**